Sources/NIOConcurrencyHelpers/lock.swift

```diff
@@ -252,12 +252,16 @@ public final class ConditionLock<T: Equatable> {
         var curTime = timeval()
         gettimeofday(&curTime, nil)
 
+	#if canImport(wasi_pthread)
+	typealias _Seconds = Int64
+	#elseif canImport(Glibc)
+	typealias _Seconds = time_t
+	#else
+	typealias _Seconds = Int
+	#endif
+
         let allNSecs: Int64 = timeoutNS + Int64(curTime.tv_usec) * 1000
-        #if canImport(wasi_pthread)
-        let tvSec = curTime.tv_sec + (allNSecs / nsecPerSec)
-        #else
-        let tvSec = curTime.tv_sec + Int((allNSecs / nsecPerSec))
-        #endif
+        let tvSec = curTime.tv_sec + _Seconds((allNSecs / nsecPerSec))
```

Comment:
I'm inclined to assume that we could actually just make this `time_t` on all platforms.